src/mumble/main.cpp

```diff
@@ -553,6 +558,32 @@ int main(int argc, char **argv) {
 	google::protobuf::ShutdownProtobufLibrary();
 #endif
 #endif
+	
+	// At this point termination of our process is immenent. We can safely
+	// launch another version of Mumble. The reason we do an actual
+	// restart instead of re-creating our data structures is that making
+	// sure we won't leave state is quite tricky. Mumble has quite a
+	// few spots which might not consider seeing to basic initializations.
+	// Until we invest the time to verify this, rather be safe (and a bit slower)
+	// than sorry (and crash/bug out). Also take care to reconnect if possible.
+	if (res == MUMBLE_EXIT_CODE_RESTART) {
+		QStringList arguments;
```

Comment:
We would have to filter out the URL, the program name and possible other stuff. Seemed simpler to just re-create it. This patch adds the url we extract from the current connection when shutting down so it isn't the one passed on startup (I just reused the variable as it was there). Basically the goal is to get as close to the state before shutdown with as little effort as possible. Thanks to the desired channel settings the client should auto-reconnect to the server and channel the user was residing on before.

Now that I look at it maybe we should even be taking the suppress identity from the global state instead of last command-line in case the user set it in the config. Not sure if the behavior would be identical though.